### PR TITLE
Support bsd operating systems

### PR DESCRIPTION
**Pull request by Martinfx**
* Fixed failed build with cmake for FreeBSD
* FIxed headers for BSD OS

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -203,7 +207,9 @@ endif()
 # builds without need for -fno-omit-frame-pointer
 set_source_files_properties("${CMAKE_CURRENT_SOURCE_DIR}/src/openloco/interop/interop.cpp" PROPERTIES COMPILE_FLAGS -fno-omit-frame-pointer)
 
-set(CMAKE_CXX_STANDARD 17)
+if (NOT CMAKE_CXX_STANDARD)
```
Comment (janisozaur) [inline CMakeLists.txt]: What's the reason for this change? I would prefer this not be set externally, the project relies on c++17

File: src/openloco/platform/platform.posix.cpp [hunk 2]
```diff
@@ -13,6 +13,12 @@
 #include <unistd.h>
 #endif
 
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__DragonFly__)
+#include <sys/sysctl.h>
```
Comment (janisozaur) [inline src/openloco/platform/platform.posix.cpp]: Might've been more useful to use c++17's `has_include` to automatically support any other systems that would require those

File: CMakeLists.txt [hunk 3]
```diff
@@ -187,7 +186,12 @@ endif()
 find_package(SDL2_mixer REQUIRED)
 
 # The hint provided here is targetting Arch Linux, a distro of choice for many contributors
-find_package(yaml-cpp REQUIRED HINTS /usr/lib32/cmake/yaml-cpp)
+if ("${CMAKE_SYSTEM_NAME}" MATCHES "(Free|Net|Open|DragonFly)BSD")
+    find_package(yaml-cpp REQUIRED)
+    include_directories(${YAML_CPP_INCLUDE_DIR})
```
Comment (janisozaur) [inline CMakeLists.txt]: This is odd. I assume the include directories were not propagated properly in previous version, but that should be addressed for all the cases, not just those select BSDs

File: CMakeLists.txt (hunk 1 above)
Comment (Martinfx) [reply CMakeLists.txt]: your compiler must not support for C++17

File: src/openloco/platform/platform.posix.cpp (hunk 2 above)
Comment (Martinfx) [reply src/openloco/platform/platform.posix.cpp]: FreeBSD have default compilet clang. I use clang compiler.

File: CMakeLists.txt (hunk 3 above)
Comment (Martinfx) [reply CMakeLists.txt]: And where issue ?

File: CMakeLists.txt (hunk 1 above)
Comment (Martinfx) [reply CMakeLists.txt]: yes

File: src/openloco/platform/platform.posix.cpp (hunk 2 above)
Comment (Martinfx) [reply src/openloco/platform/platform.posix.cpp]: but this changes are correct

Comment (janisozaur) [reply src/openloco/platform/platform.posix.cpp]: I did not say they are incorrect, just that they could be improved still.

File: CMakeLists.txt (hunk 3 above)
Comment (janisozaur) [reply CMakeLists.txt]: The issue is how you split the code, where it should not be required

File: CMakeLists.txt (hunk 1 above)
Comment (janisozaur) [reply CMakeLists.txt]: Where is your `CMAKE_CXX_STANDARD` set then? This is up to the project to set the required standard, not user.
